### DISCUSSION TITLE
Toggle format_on_save()

### DISCUSSION THREAD

**Discussion by jborbik** [VonHeikemen/lsp-zero.nvim#254] [Q&A] [answered]
I have a pretty niche question. I would like to use a global variable such as `vim.g.autoformat` which would decide whether the autoformat previously set up with `lsp.format_on_save()` should execute. Is there an easy way to achieve that with lsp-zero.nvim?

I will understand if it’s out of scope for lsp-zero.nvim.

**Comment by VonHeikemen** [comment, marked as answer]
There is a variable that controls this, is buffer local variable . If you set it to zero it will disable autoformat in that buffer.

You could do something like this.

```lua
lsp.on_attach(function(client, bufnr)
  lsp.default_keymaps({buffer = bufnr})

  if some_condition() then
    vim.b.lsp_zero_enable_autoformat = 0
  end
end)
```

(Eventually I'll add a command to toggle autoformat).

**Reply by jborbik** [reply]
That’s great! Thank you so much!
